PICMI_Python/base.py

```diff
@@ -51,3 +53,73 @@ def init(self, kw):
         # --- The implementation of this routine should use kw.pop() to retrieve input arguments from kw.
         # --- This allows testing for any unused arguments and raising an error if found.
         pass
+
+    def _check_unsupported_argument(self, arg_name, extra_info=None, raise_error=False):
+        """Raise a warning or exception if an unsupported argument was specified by the user
+         - arg_name: The name of the unsupported argument (string)
+         - extra_info: Extra information to include in the warning/error message (string)
+         - raise_error: If False (the default), raise a warning. If true, raise an exception
+                        (which interrupts the code).
+        """
+        # --- Implementation note: This should be called in the "init" method of the implementing
+        # --- class for each unsupported argument. For example, for the 'density_scale' argument of Species:
+        # ---     self._check_unsupported_argument('density_scale', extra_info='My code can not handle a density_scale')
```

Comment:
Shall we move this in the regular doc string?
```suggestion
        """Raise a warning or exception if an unsupported argument was specified by the user

         Implementation note: This should be called in the "init" method of the
         implementing class for each unsupported argument. For example, for the
         'density_scale' argument of Species:
             self._check_unsupported_argument(
                 'density_scale',
                 extra_info='My code can not handle a density_scale')

         - arg_name: The name of the unsupported argument (string)
         - extra_info: Extra information to include in the warning/error message (string)
         - raise_error: If False (the default), raise a warning. If true, raise an exception
                        (which interrupts the code).
        """
```